src/main/java/com/exadel/movie_new/controller/MovieController.java

```diff
@@ -1,48 +1,103 @@
+        if (!isNumeric(movie.getId())) {
+            throw new IdNotValidException("could not add a movie without valid id");
+        }
+        return new ResponseEntity<>(movieService.addMovie(movie), HttpStatus.CREATED);
     }
 
     @GetMapping("/{movieId}")
-    public Movie getMovie(@PathVariable String movieId) {
+    public ResponseEntity<Movie> getMovie(@PathVariable String movieId) {
 
-        return movieService.getMovie(movieId);
+        if (!isNumeric(movieId)) {
+            throw new IdNotValidException("you have to provide a valid Id");
+        }
+        if (movieService.getMovie(movieId) == null) {
+            throw new MovieNotFoundException("requested movie not found");
+        }
+
+        try {
+            return new ResponseEntity<>(movieService.getMovie(movieId), HttpStatus.OK);
+        } catch (Exception e) {
+            throw new GeneralException("something went wrong" + e.getMessage());
+        }
     }
 
-    @GetMapping()
-    public List<Movie> getAllMovies() {
+    @GetMapping
+    public ResponseEntity<List<Movie>> getAllMovies() {
+
+
+        if (movieService.getAllMovies().isEmpty()) {
+            throw new MovieNotFoundException("no movies found");
+        }
 
-        return movieService.getAllMovies();
+        try {
+            return new ResponseEntity<>(movieService.getAllMovies(), HttpStatus.OK);
+        } catch (Exception e) {
+            throw new GeneralException("something went wrong" + e.getMessage());
+        }
     }
-    
+
     @PutMapping("/{movieId}")
-    public Movie updateMovie(@RequestBody Movie movie) {
-        return movieService.updateMovie(movie);
+    public ResponseEntity<Movie> updateMovie(@RequestBody Movie movie) {
+        if (!isNumeric(movie.getId())) {
+            throw new IdNotValidException("you have to provide a valid Id");
+        }
+        if (movieService.getMovie(movie.getId()) == null) {
+            throw new MovieNotFoundException("requested movie not found");
+        }
+
+        try {
+            return new ResponseEntity<Movie>(movieService.updateMovie(movie), HttpStatus.CREATED);
+        } catch (Exception e) {
+            throw new GeneralException("something went wrong" + e.getMessage());
+        }
     }
 
     @DeleteMapping("/{movieId}")
-    public void deleteMovie(@PathVariable String movieId) {
-        movieService.deleteMovie(movieId);
-    }
+    public ResponseEntity<Void> deleteMovie(@PathVariable String movieId) {
 
+        if (!isNumeric(movieId)) {
+            throw new IdNotValidException("you have to provide a valid Id");
+        }
+        if (movieService.getMovie(movieId) == null) {
+            throw new MovieNotFoundException("the movie you are trying to delete does not exists");
+        }
+
+        try {
+            movieService.deleteMovie(movieId);
+            return new ResponseEntity<Void>(HttpStatus.ACCEPTED);
+        } catch (Exception e) {

```

Comment:
If you have ExceptionHandler - please remove **all**  "try->catch" from controller. Local Exceptions like MovieNotFoundException must be in the service layer but should be handled by ExceptionHandler and transformed into ErrorInfo